File: packages/domain/src/combinatorics.ts
export function combination(n: number, r: number): number {
  if (r < 0 || n < 0 || r > n) {
    return 0;
  }
  const selected = Math.min(r, n - r);
  let result = 1;
  for (let index = 1; index <= selected; index += 1) {
    result = Math.floor((result * (n - selected + index)) / index);
  }
  return result;
}

export function calculateBets(
  primaryCount: number,
  secondaryCount: number,
  primaryPick: number,
  secondaryPick: number,
): number {
  return combination(primaryCount, primaryPick) * combination(secondaryCount, secondaryPick);
}

File: server/app/domain/combinatorics.py
def combination(n: int, r: int) -> int:
    if r < 0 or n < 0 or r > n:
        return 0
    selected = min(r, n - r)
    result = 1
    for i in range(1, selected + 1):
        result = result * (n - selected + i) // i
    return result


def calculate_bets(
    primary_count: int,
    secondary_count: int,
    primary_pick: int,
    secondary_pick: int,
) -> int:
    return combination(primary_count, primary_pick) * combination(
        secondary_count,
        secondary_pick,
    )

